src_python/fl_server.py

```diff
@@ -0,0 +1,259 @@
+"""
+Copyright 2020 JasmineGraph Team
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+import socket
+import pickle
+import select
+import time
+import numpy as np
+import pandas as pd
+import sys
+import logging
+from timeit import default_timer as timer
+import gc
+import math
+
+logging.basicConfig(
+    level=logging.INFO, 
+    format='%(asctime)s : [%(levelname)s]  %(message)s',
+    handlers=[
+        logging.FileHandler('server.log'),
+        logging.StreamHandler(sys.stdout)
+    ]
+)
+
+class Server:
+
+    def __init__(self, MODEL, ROUNDS , weights_path, graph_id, MAX_CONN = 2, IP= socket.gethostname(), PORT = 5000, HEADER_LENGTH = 10 ):
+
+        # Parameters
+        self.HEADER_LENGTH =  HEADER_LENGTH
+        self.IP = IP
+        self.PORT = PORT
+        self.MAX_CONN = MAX_CONN
+        self.ROUNDS = ROUNDS
+
+        self.weights_path = weights_path
+        self.graph_id = graph_id
+
+        # Global model
+        self.GLOBAL_WEIGHTS = MODEL
+
+        self.global_modlel_ready = False
+
+        self.weights = []
+        self.partition_sizes = []
+        self.training_cycles = 0
+
+        self.stop_flag = False
+
+        # List of sockets for select.select()
+        self.sockets_list = []
+        self.clients = {}
+        self.client_ids = {}
+
+        # Craete server socket
+        self.server_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        self.server_socket.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+        self.server_socket.bind((self.IP, self.PORT))
+        self.server_socket.listen(self.MAX_CONN)
+
+        self.sockets_list.append(self.server_socket)
+
+    def update_model(self,new_weights,num_examples):
```

Comment:
Here the function parameters do not match with the function comment. Please check the comment and update it.

Reply:
Done